File: frontend/src/types/vault.ts
export type VaultIcon =
  | "shield"
  | "briefcase"
  | "user"
  | "credit-card"
  | "globe"
  | "server"
  | "heart"
  | "star";

export interface Vault {
  id: string;
  name: string;
  description?: string;
  icon: VaultIcon;
  itemCount: number;
  createdAt: string;
  updatedAt: string;
}

export type VaultItemType = "login" | "card" | "note" | "identity";

export interface VaultItem {
  id: string;
  vaultId: string;
  type: VaultItemType;
  name: string;
  username?: string;
  password?: string;
  url?: string;
  notes?: string;
  /** Free-form labels ("2fa", "family", …) — surfaced on the identity map. */
  tags?: string[];
  favorite: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateVaultPayload {
  name: string;
  description?: string;
  icon: VaultIcon;
}

export type UpdateVaultPayload = Partial<CreateVaultPayload>;

export interface CreateVaultItemPayload {
  vaultId: string;
  type: VaultItemType;
  name: string;
  username?: string;
  password?: string;
  url?: string;
  notes?: string;
  favorite?: boolean;
}

export type UpdateVaultItemPayload = Partial<
  Omit<CreateVaultItemPayload, "vaultId">
>;

File: backend/app/schemas/vault.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime, time

class CreateVaultPayload(BaseModel):
    name: str  # maps to encrypted_name
    description: str | None = None
    icon: str | None = None

class UpdateVaultPayload(BaseModel):
    name: str | None = None

class VaultOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str  # maps from encrypted_name
    itemCount: int = 0
    createdAt: str
    updatedAt: str
